### PR TITLE
Add CTCI 2.2 (Kth to Last) for Linked List session

### LINKED ISSUE #81: Week 2 Linked Lists: Write-up for question 1
Procedure for completing a problem & solution writeup for UWB ACM CTCI sessions:

- [x] Assign an issue to yourself
- [x] Pick a problem to write up. Please check the prior quarter's session to ensure we don't have duplicates between quarters. Some good places to look are [LeetCode](https://leetcode.com/), [HackerRank](https://www.hackerrank.com/), and of course [CTCI](http://www.crackingthecodinginterview.com). Communicate with your colleagues to ensure we have good topic coverage and nobody is writing up duplicate questions.
- [x] Clone this repository to your computer, if you haven't already. If you have, `cd` into the repo's directory, run `git checkout master`, and run `git pull origin master`.
- [x] Run `git checkout -b <my-topic-name>`, and name the branch something logical. (Pick a short name for the problem you are writing up, for example.)
- [x] `cd` into the directory with the solutions file for the session. The naming conventions should make the file's location fairly self-evident.
- [ ] Add the problem description in the appropriate numbered selection in the solutions file.
- [ ] Add the solution writeup in the appropriate numbered selection in the solutions file.
- [ ] Add a driver program (which includes your solution) in a subdirectory that produces the output included in the solutions file.
- [ ] Run `git commit`. Add a useful commit description, so that reviewers know what they're looking at.
- [ ] Run `git push origin <my-topic-name>`, using the branch name you chose earlier.
- [ ] Navigate to [the GitHub repository](https://github.com/UWB-ACM/CTCI) and open a Pull Request for your submitted branch.
- [ ] Request a review from one or more designated PR reviewers for the session.
- [ ] If your reviewer requests changes, please address all suggestions.

As always, if you experience issues or have questions, talk to your peers. We are here to help each other!

### PR DESCRIPTION
**Pull request by etcadinfinitum**
Closes #81

### CODE REVIEW

File: 2019-03-Autumn/1_linked_lists/README.md [hunk 1]
```diff
@@ -141,21 +171,128 @@ Node* sumOrderedNums(Node* n1, Node* n2) {
...
+        return False, current
+    # We need to continue looking for our target node.
+    elif depth < k and current is not head:
+        return True, depth + 1
+    # If no previous conditions are met, we will not be able to find a match.
+    return False, None
+```
 
-#### Optimal Solution
+#### Testing The Solutions OR Driver For Solution
```
Comment (etcadinfinitum) [inline 2019-03-Autumn/1_linked_lists/README.md]:
```suggestion
#### Testing The Solutions
```

File: 2019-03-Autumn/1_linked_lists/README.md [hunk 2]
```diff
@@ -141,21 +171,128 @@ Node* sumOrderedNums(Node* n1, Node* n2) {
...
+```
 
-TODO :bug:
+#### Recursive Solution (Less Simple)
+
+Achieving the same result using recursion is more cumbersome when 
+compared to the iterative solution and requires some inventiveness on 
+our part.
+
+In theory, we should be able to traverse the entire list and increment
```
Comment (etcadinfinitum) [inline 2019-03-Autumn/1_linked_lists/README.md]:
```suggestion
In theory, we should be able to traverse the entire list and then increment 
```

File: 2019-03-Autumn/1_linked_lists/README.md [hunk 3]
```diff
@@ -141,21 +171,128 @@ Node* sumOrderedNums(Node* n1, Node* n2) {
...
-TODO :bug:
+#### Recursive Solution (Less Simple)
+
+Achieving the same result using recursion is more cumbersome when 
+compared to the iterative solution and requires some inventiveness on 
+our part.
+
+In theory, we should be able to traverse the entire list and then increment 
+a counter as the stack frames resolve until the counter reaches `k`. 
+However, this presents several problems for implementation:
```
Comment (mkhsu) [inline 2019-03-Autumn/1_linked_lists/README.md]: stack frames 'resolve' kinda sounds weird

File: 2019-03-Autumn/1_linked_lists/README.md [hunk 4]
```diff
@@ -141,21 +171,128 @@ Node* sumOrderedNums(Node* n1, Node* n2) {
...
 
-TODO :bug:
+#### Recursive Solution (Less Simple)
+
+Achieving the same result using recursion is more cumbersome when 
+compared to the iterative solution and requires some inventiveness on 
+our part.
+
+In theory, we should be able to traverse the entire list and then increment 
+a counter as the stack frames resolve until the counter reaches `k`.
```
Comment (etcadinfinitum) [inline 2019-03-Autumn/1_linked_lists/README.md]:
```suggestion
a counter as each recursive call returns until the counter reaches `k`. 
```

File: 2019-03-Autumn/1_linked_lists/README.md (hunk 3 above)
Comment (etcadinfinitum) [reply 2019-03-Autumn/1_linked_lists/README.md]: Thank ye

**Pull request merged**